storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/KvStoreDatabase.java

```diff
@@ -303,6 +304,25 @@ public List<SignedBeaconBlock> getNonCanonicalBlocksAtSlot(final UInt64 slot) {
   }
 
   protected void storeFinalizedBlocksToDao(
+      final Collection<SignedBeaconBlock> blocks,
+      final Map<UInt64, List<BlobSidecar>> finalizedBlobSidecarsBySlot) {
+    try (final FinalizedUpdater updater = finalizedUpdater()) {
+      blocks.forEach(
+          block -> {
+            updater.addFinalizedBlock(block);
+            final List<BlobSidecar> blobSidecars =
+                finalizedBlobSidecarsBySlot.getOrDefault(block.getSlot(), Collections.emptyList());
+            if (blobSidecars.isEmpty()) {
+              updater.addNoBlobsSlot(new SlotAndBlockRoot(block.getSlot(), block.getRoot()));
```

Comment:
I am not sure of the implications of this since this will start to be persisted pre-Deneb. Is there an overhead of doing this operation. Maybe could be an idea to make the map an optional pre-Deneb?

Reply:
should be fixed with https://github.com/ConsenSys/teku/pull/7028/commits/81d070b1662df0cd87b3da93813a5f610cd237b1

Reply:
not that simple
with slots without blocks